### PR TITLE
DefaultMethodHandler performance: cache MethodHandles.Lookup instance…

### PR DESCRIPTION
**Pull request by qualidafial**
…s per declaring class, and method handles per default method handler.

### CODE REVIEW

**Comment by qualidafial** [comment]
@arteam Would you please review this?

**Comment by arteam** [comment]
LGTM

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java [hunk 1]
```diff
@@ -13,55 +13,65 @@
...
 
 import org.jdbi.v3.core.HandleSupplier;
 
-class DefaultMethodHandler implements Handler
-{
-    DefaultMethodHandler()
-    {
-    }
+class DefaultMethodHandler implements Handler {
+    private static final Map<Class<?>, MethodHandles.Lookup> privateLookups = new ConcurrentHashMap<>();
```
Comment (stevenschlansker) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java]: Since this is static, this likely breaks class unloading.

Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java]: You're right. I'll submit a follow-up that uses weak keys.

Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java]: Looks like `SqlObjectFactory` is also caching classes in `ConcurrentHashMap`.

Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java]: Added card https://github.com/jdbi/jdbi/projects/1#card-304665 to project board

**Pull request merged**